src/CMakeLists.txt

```diff
@@ -33,8 +35,25 @@ if(CMAKE_EXTRA_GENERATOR OR MSVC_IDE)
     list(APPEND CUKE_SOURCES ${CUKE_HEADERS})
 endif()
 
-add_library(cucumber-cpp-nomain STATIC ${CUKE_SOURCES})
-add_library(cucumber-cpp STATIC ${CUKE_SOURCES} main.cpp)
+if(CUKE_ENABLE_SHARED_LIB)
+    set(CUKE_LINKAGE SHARED)
+else()
+    set(CUKE_LINKAGE STATIC)
+endif()
+
+add_library(cucumber-cpp-nomain ${CUKE_LINKAGE} ${CUKE_SOURCES})
+add_library(cucumber-cpp ${CUKE_LINKAGE} ${CUKE_SOURCES} main.cpp)
+
+set_target_properties(cucumber-cpp cucumber-cpp-nomain PROPERTIES
```

Comment:
I'm actually not sure why this is here anymore; IIRC it was added by request of the upstream maintainers.